Framework/LiveData/test/KafkaTesting.h

```diff
@@ -387,6 +388,125 @@ class FakeRunInfoStreamSubscriber
   int32_t m_nperiods = 1;
 };
 
+// -----------------------------------------------------------------------------
+// Fake run data stream with incrementing number of periods
+// -----------------------------------------------------------------------------
+class FakeRunInfoStreamSubscriberVaryingNPeriods
+    : public Mantid::LiveData::IKafkaStreamSubscriber {
+public:
+  void subscribe() override {}
+  void subscribe(int64_t offset) override { UNUSED_ARG(offset) }
+  void consumeMessage(std::string *buffer, int64_t &offset, int32_t &partition,
+                      std::string &topic) override {
+    assert(buffer);
+
+    fakeReceiveARunStartMessage(buffer, m_runNumber, m_startTime, m_instName,
+                                m_nperiods);
+
+    m_nperiods++;
+    m_runNumber++;
+    UNUSED_ARG(offset);
+    UNUSED_ARG(partition);
+    UNUSED_ARG(topic);
+  }
+  std::unordered_map<std::string, std::vector<int64_t>>
+  getOffsetsForTimestamp(int64_t timestamp) override {
+    UNUSED_ARG(timestamp);
+    return {
+        std::pair<std::string, std::vector<int64_t>>("topic_name", {1, 2, 3})};
+  }
+  std::unordered_map<std::string, std::vector<int64_t>>
+  getCurrentOffsets() override {
+    std::unordered_map<std::string, std::vector<int64_t>> offsets;
+    return offsets;
+  }
+  void seek(const std::string &topic, uint32_t partition,
+            int64_t offset) override {
+    UNUSED_ARG(topic);
+    UNUSED_ARG(partition);
+    UNUSED_ARG(offset);
+  }
+
```

Comment:
Probably best to have a line spacing between each of the methods that have been implemented here? I think it would make the code a bit easier to read...